src/components/NewAnalysisModal.js

```diff
@@ -64,33 +139,120 @@ export const NewAnalysisModal = Utils.withDisplayName('NewAnalysisModal')(
+      renderCreateAnalysisBody(currentTool)
+    ])
+
+    const existingNames = _.map(({ name }) => getDisplayName(name), analyses)
+
+    const errors = validate(
+      { analysisName, notebookKernel },
+      {
+        analysisName: analysisNameValidator(existingNames),
+        notebookKernel: { presence: { allowEmpty: true } }
+      }
+    )
+
+    const renderCreateAnalysisBody = toolLabel => div({ style: { display: 'flex', flexDirection: 'column' } }, [
+      h(IdContainer, [id => h(Fragment, [
+        h(FormLabel, { htmlFor: id, required: true }, [`Name of the ${getArtifactLabel(toolLabel)}`]),
+        analysisNameInput({
+          error: Utils.summarizeErrors(nameTouched && errors?.analysisName),
+          inputProps: {
+            id, value: analysisName,
+            onChange: v => {
+              setAnalysisName(v)
+              setNameTouched(true)
+            }
+          }
+        })
+      ])]),
+      toolLabel === tools.Jupyter.label && h(IdContainer, [id => h(Fragment, [
+        h(FormLabel, { htmlFor: id, required: true }, ['Language']),
+        h(Select, {
+          id, isSearchable: true,
+          placeholder: 'Select a language',
+          getOptionLabel: ({ value }) => _.startCase(value),
+          value: notebookKernel,
+          onChange: ({ value: notebookKernel }) => setNotebookKernel(notebookKernel),
+          options: ['python3', 'r']
+        })
+      ])]),
+      (toolLabel === tools.Jupyter.label || toolLabel === tools.RStudio.label) && (currentRuntime && !isRuntimeDeletable(currentRuntime) && currentRuntimeTool !== toolLabel) && div({ style: { backgroundColor: colors.warning(0.1), margin: '.5rem', padding: '1rem' } }, [
+        h(WarningTitle, { iconSize: 16 },
+          [span({ style: { fontWeight: 600 } }, ['Environment Creation'])]
+        ),
+        div({ style: { marginBottom: '.5rem', marginTop: '1rem' } }, ['You have a non-deletable environment associated with another application.']),
+        div(['You may create an analysis, but must wait for your current environment to finish processing and get a suitable environment to run it.'])
+      ]),
+      div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '1rem' } }, [
+        h(ButtonPrimary, {
+          disabled: errors,
+          tooltip: Utils.summarizeErrors(errors),
+          onClick: async () => {
+            try {
+              const contents = toolLabel === tools.Jupyter.label ? notebookData[notebookKernel] : '# Starter Rmd file'
+              if (toolLabel === tools.Jupyter.label) {
+                // notebook creation requires special handling of json in request
+                await Ajax().Buckets.notebook(namespace, bucketName, analysisName).create(contents)
+              } else {
+                await Ajax().Buckets.analysis(namespace, bucketName, analysisName, toolLabel).create(contents)
+              }
```

Comment:
Could you make this a ternary?

Reply:
I thought the syntax was a bit odd with awaiting in a ternary... i do think it looks cleaner though